frontend/src/app/utils/internalChartHelper.ts

```diff
@@ -963,3 +964,148 @@ export const variableToFilter = (
     };
   });
 };
+
+// NOTE There are functions from Echarts, but there are changes.
+// NOTE from echarts/src/scale/Interval.ts.
+export function calcNiceExtent(
```

Comment:
why not import directly from echarts/src/interval.ts, then export from chart util?